### PR TITLE
crypto: Create intent_signing.md

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Add a doc here about intent signing. Before we have the new doc site, this is a place to refer to for any intent signing related questions. 

## Test Plan 

N/A
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/intent.rs [hunk 1]
```diff
@@ -58,15 +58,18 @@ impl Default for AppId {
...
-    CheckpointSummary = 2,
-    PersonalMessage = 3,
-    SenderSignedTransaction = 4,
-    ProofOfPossession = 5,
+    TransactionData = 0, // Used for a user signature on a transaction data.
+    TransactionEffects = 1, // Used for an authority signature on transaction effects.
+    CheckpointSummary = 2, // Used for an authority signature on a checkpoint summary.
+    PersonalMessage = 3, // Used for a user signature on a personal message.
+    SenderSignedTransaction = 4, // Used for an authority signature on a user signed transaction. 
+    ProofOfPossession = 5, // Used for the authority signature used as a proof of possession.
```
Comment (kchalkias) [inline crates/sui-types/src/intent.rs:72]: better message

File: crates/sui/intent_signing.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const intentMessage = messageWithIntent(
+    IntentScope.TransactionData,
+    transactionBytes,
+);
+const signature = await this.signData(intentMessage);
+```
+
+## Authority Signature
+
+Authority signature is created using the protocol key. The data that it commits to is also an intent message `intent || message`. See all available intent scope [here](https://github.com/MystenLabs/sui/blob/0dc1a38f800fc2d8fabe11477fdef702058cf00d/crates/sui-types/src/intent.rs#L66)
```
Comment (kchalkias) [inline crates/sui/intent_signing.md:74]: scope**s**

File: crates/sui/intent_signing.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+The IntentMessage struct consists of the intent itself and the serialized data value.
+
+```rust
+pub struct IntentMessage<T> {
+    pub intent: Intent,
+    pub value: T,
+}
+```
+
+An intent is a compact struct serves as the domain separator for a message that a signature commits to. It consists of three parts: `IntentScope` (what the type of the message is), `IntentVersion` (what version the network supports), `AppId` (what application that the signature refers to).
```
Comment (kchalkias) [inline crates/sui/intent_signing.md:34]: a compact struct **that** serves

**Comment by joyqvq** [comment]
todo: add a section on why we dont send the intent byte over the wire but let fullnode to attach
```
the idea here is the client should not need to submit the three extra byte for intent because  1) saves a little bandwidth on what to send, we still have the sig commits to the intent, but no need to send over the wire. 2)  fullnode should know by api that what the intent is. currently, this endpoint only accepts IntentScope::TransactionData and AppId::Sui. If one day we want to bump IntentVersion::V0 -> V1 for fullnode,  we attach intent bytes here to enforce that the sig is verified against v1. the client needs to decide to create valid sig on v1.
```

**Review by kchalkias — APPROVED** [review]
LGTM

**Pull request merged**